wordpress-mobile/WordPress-Android
# Extract auto save functionality

Fixes #10693 
This PR extracts the auto save logic into a ViewModel and adds some unit tests. The functionality should not have changed. 

To test:
* Test the Editor (mostly saving/autosaving posts and drafts) and check that there is no functionality change

PR submission checklist:
